code/game/objects/effects/turf_fire.dm

```diff
@@ -0,0 +1,183 @@
+
+/obj/effect/abstract/turf_fire/inferno
+	fire_power = 30
+
+/obj/effect/abstract/turf_fire/inferno/magical
+	magical = TRUE
+
+/obj/effect/abstract/turf_fire/Initialize(mapload, power)
+	. = ..()
+	var/turf/open/open_turf = loc
+	if(open_turf.turf_fire)
+		return INITIALIZE_HINT_QDEL
+	/* for crossed removal
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = .proc/on_entered,
+	)
+	AddElement(/datum/element/connect_loc, src, loc_connections)
+	*/
+	open_turf.turf_fire = src
+	SSturf_fire.fires[src] = TRUE
+	if(power)
+		fire_power = min(TURF_FIRE_MAX_POWER, power)
+	UpdateFireState()
+
+/obj/effect/abstract/turf_fire/Destroy()
+	var/turf/open/open_turf = loc
+	open_turf.turf_fire = null
+	SSturf_fire.fires -= src
+	return ..()
+
+/obj/effect/abstract/turf_fire/proc/process_waste()
+	var/turf/open/open_turf = loc
+	if(open_turf.planetary_atmos)
+		return TRUE
+	var/datum/gas_mixture/cached_air = open_turf.air
+	var/oxy = cached_air.get_moles(/datum/gas/oxygen)
+	if (oxy < 0.5)
+		return FALSE
+	var/temperature = cached_air.return_temperature()
+	var/old_heat_capacity = cached_air.heat_capacity()
+	var/burn_rate = TURF_FIRE_BURN_RATE_BASE + fire_power * TURF_FIRE_BURN_RATE_PER_POWER
+	if(burn_rate > oxy)
+		burn_rate = oxy
+
+	var/new_o2 = (cached_air.get_moles(/datum/gas/oxygen) - burn_rate)
+	cached_air.set_moles(/datum/gas/oxygen, new_o2)
+
+	var/new_co2 = (cached_air.get_moles(/datum/gas/carbon_dioxide) + burn_rate * TURF_FIRE_BURN_CARBON_DIOXIDE_MULTIPLIER)
+	cached_air.set_moles(/datum/gas/carbon_dioxide, new_co2)
+
+	var/new_heat_capacity = cached_air.heat_capacity()
+	var/energy_released = burn_rate * TURF_FIRE_ENERGY_PER_BURNED_OXY_MOL
+	cached_air.adjust_heat((temperature * old_heat_capacity + energy_released) / new_heat_capacity)
+	open_turf.air_update_turf(TRUE)
+	return TRUE
+
+/obj/effect/abstract/turf_fire/process()
+	var/turf/current_turf = loc
+	if(!isopenturf(current_turf)) //This can happen, how I'm not sure
+		qdel(src)
+		return
+	var/turf/open/open_turf = loc
+	if(open_turf.active_hotspot) //If we have an active hotspot, let it do the damage instead and lets not loose power
+		return
+	if(!magical)
+		if(!process_waste())
+			qdel(src)
+			return
+		if(open_turf.air.return_temperature() < TURF_FIRE_REQUIRED_TEMP)
+			fire_power -= TURF_FIRE_POWER_LOSS_ON_LOW_TEMP
+		fire_power--
+		if(fire_power <= 0)
+			qdel(src)
+			return
+	open_turf.hotspot_expose(TURF_FIRE_TEMP_BASE + (TURF_FIRE_TEMP_INCREMENT_PER_POWER*fire_power), TURF_FIRE_VOLUME)
+	for(var/A in open_turf)
+		var/atom/AT = A
+		AT.fire_act(TURF_FIRE_TEMP_BASE + (TURF_FIRE_TEMP_INCREMENT_PER_POWER*fire_power), TURF_FIRE_VOLUME)
+	if(!magical)
+		if(prob(fire_power))
+			open_turf.burn_tile()
+		if(prob(6))
+			playsound(open_turf, 'sound/effects/comfyfire.ogg', 40, TRUE)
+		UpdateFireState()
+
+// /obj/effect/abstract/turf_fire/proc/on_entered(datum/source, atom/movable/AM) for crossed removal
+/obj/effect/abstract/turf_fire/Crossed(atom/movable/AM)
```

Comment:
descript variable names

Reply:
Wouldn't it be better to keep it how it's named in the parent proc? Testing shows that it works when you change it to something more readable